hw1/02-palindrome.js

```diff
@@ -1,3 +1,36 @@
-const elem = document.querySelector('input');
+//returns true if the given number is a palindrome, false otherwise
+const isPalindrome = (num) => {
+  const original = num.toString();
 
-elem.addEventListener('input', handleInput);
+  //reverse string by splitting it into an array and reversing that, then rejoining
+  //credit: https://www.freecodecamp.org/news/how-to-reverse-a-string-in-javascript-in-3-different-ways-75e4763c68cb/
+  const reversed = original.split("").reverse().join("");
+
+  return reversed === original;
+};
+
+//Display to user whether input was a palindrome or not
+// displays invalid input if number is negative
+const handleInput = () => {
+  const input = document.getElementById("numberInput").value;
+  let outputElement = document.getElementById("response");
+  let responseText;
+
+  //Determine response to display, error on negative numbers and NaN/empty input
+  if (input === NaN || input === "" || input < 0) {
+    responseText = "Please enter a positive number";
+    outputElement.className = "text-danger";
+  } else if (isPalindrome(input)) {
+    responseText = "Yes. This is a palindrome!";
+    outputElement.className = "text-success";
+  } else {
+    responseText = "No. Try again.";
+    outputElement.className = "text-danger";
+  }
+
+  outputElement.textContent = responseText;
+};
+
+const elem = document.querySelector("input");
+
+elem.addEventListener("input", handleInput);
```

Comment:
You have declared the variable elem for the input element but it does not appear in your snippet.

If you meant to use it within the handleInput function, it would be best to utilize it properly or avoid declaring it if it's not used.

While your comments are generally helpful, ensure they add value and aren't merely describing what the code does, which can be apparent from the code itself. Comments are best used to explain "why" something is done, not "how".

Reply:
I agree with your statement about comments with the caveat that I personally think functions should have rather thorough comment describing the details of what they do. This may be due to my general preference for strongly typed languages, but having a quick reference of exactly what types a function expects and returns is something I find useful.

Having said that, I have updated my comments to abide by the general criteria:
- If it's before a function, it describes what the function does in full detail
- Otherwise, it should be explaining why the code does things in the way that it does

Thanks for the feedback!